src/screens/Watch/Components/Transcriptions/CaptionLine/index.js

```diff
@@ -1,140 +1,233 @@
+    }
+    return true;
   };
 
-  const handleBlur = () => {
-    ref.current.innerText = text;
-    transControl.handleBlur();
+  const handleCancel = () => {
+    textRef.current.innerText = text;
+    timeRef.current.innerText = prettierTimeStr(fullBeginTime);
+    endTimeRef.current.innerText = prettierTimeStr(fullEndTime);
+    dispatch({ type: 'watch/setCurrEditing', payload: null });
   };
 
-  const handleSave = () => {
-    dispatch({ type: 'watch/saveCaption', payload: { caption, text: ref.current.innerText } })
+  const handleSave = (updatedBeginTime = fullBeginTime, updatedEndTime = fullEndTime) => {
+    const newText = textRef.current.innerText;
+  
+    try {
+      // eslint-disable-next-line no-console
+      console.log("Handle save triggered");
+      // eslint-disable-next-line no-console
+      console.log("Using begin time:", updatedBeginTime);
+      // eslint-disable-next-line no-console
+      console.log("Using end time:", updatedEndTime);
+  
+      validateTimeFormat(updatedBeginTime);
+      validateTimeFormat(updatedEndTime);
+      dispatch({
+        type: 'watch/saveCaption',
+        payload: { caption, text: newText, begin: updatedBeginTime, end: updatedEndTime },
+      });
+      setTimeString(prettierTimeStr(updatedBeginTime));
+      setEndTimeString(prettierTimeStr(updatedEndTime));
+      // eslint-disable-next-line no-console
+      console.log("Time strings updated to:", prettierTimeStr(updatedBeginTime), prettierTimeStr(updatedEndTime));
+      textRef.current.innerText = newText;
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error("Error during save:", error.message);
+      alert('Please enter a valid time format (HH:MM:SS, MM:SS, or HH:MM:SS.SSS)');  // eslint-disable-line no-alert
+    }
   };
 
-  const handleCancel = () => {
-    ref.current.innerText = text;
-    dispatch({ type: 'watch/setCurrEditing', payload: null })
+  const handleBlur = (ref, originalValue, setFullTime) => {
+    if (ref.current) {
+      const elementId = ref.current?.id || "";
+      const currentValue = ref.current.innerText;
+      if (elementId.includes("time")) {
+        // time
+        try {
+          validateTimeFormat(currentValue); 
+          setFullTime((prev) => {
+            // eslint-disable-next-line no-console
+            console.log("Updating time:", currentValue);
+            // eslint-disable-next-line no-console
+            console.log(`${prev}`);
+            return currentValue;
+          });
+          ref.current.innerText = prettierTimeStr(currentValue);
+        } catch {
+          // eslint-disable-next-line no-console
+          console.warn("Invalid time format. Reverting to original value:", originalValue);
+          ref.current.innerText = prettierTimeStr(originalValue);
+        }
+      } else if (elementId.includes("textarea")) {
+        // text
+        // eslint-disable-next-line no-console
+        console.log("handling blur for text");

```

Comment:
Once this code is ready for the PR the console log should be removed or commented out.